api_tester/lib/api_tests/purchases_flutter_api_test.dart

```diff
@@ -415,4 +415,11 @@ class _PurchasesFlutterApiTest {
       entitlement
     );
   }
+
+  void _checkSyncObserverModeAmazonPurchase() async {
+    Future<void> future = Purchases.syncObserverModeAmazonPurchase(
+        "productID", "receiptID", "amazonUserID", "isoCurrencyCode", 3.4);
+    future = Purchases.syncObserverModeAmazonPurchase(
+        "productID", "receiptID", "amazonUserID", null, null);
```

Comment:
Nitpick: One possible way to avoid calling this same method twice is adding the parameters as parameters of the check function with the correct type (including nullability). I think it's also more visual.